ATB/DataAccess/Repositories/Implementations/CsvFlightRepository.cs

```diff
@@ -0,0 +1,49 @@
+using DataAccess.CsvHelperService;
+using DataAccess.Models;
+using DataAccess.Repositories.Interfaces;
+using DataAccess.SearchCriteria;
+
+namespace DataAccess.Repositories.Implementations;
+
+public class CsvFlightRepository(ICsvFileService<Flight> csvFileService, string pathToCsv) : IFlightRepository
+{
+    private List<Flight> _flightsCache = null!;
+    private bool _isCacheInitialized = false;
+
+    private async Task InitializeCacheAsync()
+    {
+        if (!_isCacheInitialized)
+        {
+            _flightsCache = await csvFileService.ReadFromCsvAsync(pathToCsv);
+            _isCacheInitialized = true;
+        }
+    }
+
+    public async Task<IEnumerable<Flight>> GetAllAsync()
+    {
+        await InitializeCacheAsync();
```

Comment:
what about calling this method in the constructor instead of calling it in every method?